### PR TITLE
Add SSL support to MySQL connections via IDO

### PR DESCRIPTION
**Pull request by leeclemens**
fixes: #9725

### CODE REVIEW

File: lib/db_ido_mysql/idomysqlconnection.cpp [hunk 1]
```diff
@@ -411,7 +444,7 @@ void IdoMysqlConnection::ClearCustomVarTable(const String& table)
 	Query("DELETE FROM " + GetTablePrefix() + table + " WHERE session_token <> " + Convert::ToString(m_SessionToken));
 }
 
-void IdoMysqlConnection::ClearConfigTable(const String& table)
+void IdoMysqlConnection::ClearConfiglTable(const String& table)
```
Comment (lazyfrosch) [inline lib/db_ido_mysql/idomysqlconnection.cpp]: This seems odd, intended change?

**Comment by lazyfrosch** [comment]
Seems okay, with the exception of an (i guess) unintended change of a function name `IdoMysqlConnection::ClearConfiglTable`

**Comment by gunnarbeutner** [comment]
You also don't need to specify default values if they're empty:

    [config] String ssl_key;

File: lib/db_ido_mysql/idomysqlconnection.cpp (hunk 1 above)
Comment (leeclemens) [inline lib/db_ido_mysql/idomysqlconnection.cpp]: Not at all, Win+l to lock while the linux VM had focus, sorry about that.

**Comment by lazyfrosch** [comment]
@leeclemens you don't have to close this PR, just change, add commits, amend, rebase, whatever. and than force-push to the branch used in the PR.

**Comment by leeclemens** [comment]
Thanks @lazyfrosch , I was going to reopen it after I squashed them. It was late for me and I was having an issue testing it on a new VM, and wanted to be sure I sorted that out first :)

**Comment by leeclemens** [comment]
Sorry for the confusion...I have completed testing using `ssl_ca`, `ssl_key` and `ssl_cert` with users that `REQUIRE SSL` and `REQUIRE X509`.

From the debug statements I confirmed `ssl_capath` and `ssl_cipher` are being read correct (as well has `have_ssl`) - since we're simply passing all of these values to `mysql_ssl_set`,

I think the icinga functionality is tested (i.e. we don't really care if the cert is valid, etc...the connection error is printed). I also renamed 'has ssl' to 'have ssl', which matches MySQL's variable/syntax.

**Comment by leeclemens** [comment]
Apologies, TIL you cannot re-open a PR for a branch that was "force-pushed or recreated".
Created new PR #66

**Pull request closed without merging**